internal/dot/graph.go

```diff
@@ -130,21 +163,84 @@ func (dg *Graph) AddCtor(c *Ctor, paramList []*Param, resultList []*Result) {
 	c.Results = resultList
 
 	dg.Ctors = append(dg.Ctors, c)
+	dg.ctorMap[c.ID] = c
+}
+
+func (dg *Graph) failNode(r *Result, pof bool) {
+	if pof {
+		dg.addRootCause(r)
+	} else {
+		dg.addTransitiveFailure(r)
+	}
+}
+
+// MissingNodes adds missing nodes to the list of failed Results in the graph.
+func (dg *Graph) MissingNodes(results []*Result) {
```

Comment:
Similar to the above comment, if this is "adding" nodes, we should write `AddMissingNodes`.

Reply:
Changed to `AddMissingNodes` too.